pkg/domain/domain.go

```diff
@@ -1881,6 +1884,21 @@ func (do *Domain) LoadPrivilegeLoop(sctx sessionctx.Context) error {
 				continue
 			}
 
+			// Skip the event from this TiDB-Server
+			canSkip := true
+			for _, event := range resp.Events {
+				if event.Kv != nil {
+					val := string(event.Kv.Value)
+					if do.serverID != 0 && val != strconv.FormatUint(do.serverID, 10) {
+						canSkip = false
+						break
+					}
+				}
+			}
+			if canSkip {
+				continue
+			}
```

Comment:
I think it's better to make `canSkip` default to false and check resp to make it to true.
That's more fault-tolerant when event.Kv is not set or the values unmarshal get error. @CbcWestwolf